poc/vidpf.py

```diff
@@ -154,8 +154,19 @@ def eval(cls,
                 (seed, ctrl, y, pi_proof) = prefix_tree_share.get(
                     (node, current_level))
 
+        # Compute the Aggregator's share of `beta`.
+        y0 = prefix_tree_share[(0, 0)][2]
+        y1 = prefix_tree_share[(1, 0)][2]
+        beta_share = vec_add(y0, y1)[1:]  # first element is the counter
+        if agg_id == 1:
+            beta_share = vec_neg(beta_share)
+
+        # Compute the Aggregator's share of the counter.
+        counter_share = y0[0] + y1[0] + cls.Field(agg_id)
```

Comment:
@jimouris please check carefully that this matches PLASMA.

Reply:
You are referring to this specific line, right? `counter_share = y0[0] + y1[0] + cls.Field(agg_id)`

Which PLASMA check do you have in mind?

Reply:
> You are referring to this specific line, right? `counter_share = y0[0] + y1[0] + cls.Field(agg_id)`

Yes.

> Which PLASMA check do you have in mind?

This is the thing where we make sure the value programmed into the VIDPF (i.e., part of $\beta$) is `1`.

Reply:
Not exactly. In PLASMA we do the following checks:
* for the root level: we check that $H(1 - y_{S0}^0 - y_{S0}^1) == H(y_{S1}^0 + y_{S1}^1)$
* for all other levels: we check that $H(y_{S0}^p - y_{S0}^{p||0} - y_{S0}^{p||1}) == H(-y_{S1}^p + y_{S1}^{p||0} + y_{S1}^{p||1})$

So the check that actually checks if beta is one is the first one for the root level.

Reply:
I believe that's exactly what I'm doing here:
```python
counter_share = y0[0] + y1[0] + cls.Field(agg_id)
```
Except that the second server's share is not negated. (For some reason, the math works out this way in the current VIDPF implementation, as we noted on an earlier PR.)